### DISCUSSION TITLE
Regex in Pageview Goals path

### DISCUSSION THREAD

**Discussion by alpha077412** [plausible/analytics#241] [Anything Else]
## Prerequisites
- [x] I have searched open and closed issues to make sure that the feature has not yet been requested.

## Feature request
**Is your feature request related to a problem? Please describe.**
Currently Pageview Goal paths must be identical to those tracked in Plausible, meaning it is not possible to create goals for dynamic URLs without using custom events. For example, a Woocommerce checkout page redirects users to _/checkout/order-received/12345/_ where the final part of the URL is unique for each order. 

**How would you like it to work**
Allow entry of regex or more simply an asterisks * in Pageview Goal paths. For example, _/checkout/order-received/*_

**Update:** This is now possible! You can for instance track all the blog posts by using /blog/** (if your blog subdirectory is named blog) or track Woocommerce checkout pages for your ecommerce (/checkout/order-received/*/). full details in the docs: https://plausible.io/docs/pageview-goals

**Comment by ukutaht** [comment]
Makes sense, thanks for the request!

**Comment by youngbrioche** [comment]
Plausible should at least allow for partly URL matching (e.g. „ends with“). Would make it possible to track pageviews like

- /de/signup/thanks
- /en/signup/thanks

Without configuring two goals.

**Comment by n0th** [comment]
Yes, please, this is sooo important for dynamic pages like Woocommerce

**Comment by Vigasaurus** [comment]
Once we have https://github.com/plausible/analytics/pull/489 and its API finalized/agreed upon, I'll go through and apply the same format of path asterisk replacement for pageview goals 👍

**Reply by alpha077412** [reply]
Great stuff!

**Comment by ukutaht** [comment]
@Vigasaurus Nice. Sorry it's taken forever for me to review that PR.

**Reply by Vigasaurus** [reply]
Haha all good - I know the struggles of reviewing PRs and the lack of motivation to do it all too well 😅

**Comment by uhlhosting** [comment]
Any ETA on the regex?

**Comment by metmarkosaric** [comment]
Thanks @uhlhosting! We don't have any ETA. You can take a look at our roadmap to see what we're working on at this moment (API): https://github.com/plausible/analytics/projects/1

**Comment by Vigasaurus** [comment]
Yeah - I had decent progress on this, but I'm putting it on the side for now - ran into some walls with Clickhouse querying through regex, and then grouping correctly over the regex. I will probably do it eventually - but not actively/soon. Gonna work on the comparisons stuff first (I feel that's more widely applicable, personally).

**Comment by Vigasaurus** [comment]
👀 
![image](https://user-images.githubusercontent.com/19434157/108937745-72a87c00-7614-11eb-805d-fbc139a83539.png)

**Reply by ukutaht** [reply]
Looks nice :)

**Reply by uhlhosting** [reply]
when can we expect It in early betas?

**Reply by Vigasaurus** [reply]
It's still very early and pretty incomplete, see: https://github.com/plausible/analytics/pull/750

**Comment by metmarkosaric** [comment]
This is now live thanks to @Vigasaurus! You can for instance track all the blog posts by using `/blog/**` (if your blog subdirectory is named blog) or track Woocommerce checkout pages for your ecommerce (`/checkout/order-received/*/`). full details in the docs: https://plausible.io/docs/pageview-goals
